File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input())
array = list(map(int, input().split(' ')))
x = int(input())
array.sort()
count = 0

start = 0
end = n-1

while start != end:
    if array[start] + array[end] == x:
        count += 1
        start += 1
    elif array[start] + array[end] < x:
        start += 1
    else:
        end -= 1
print(count)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;


public class Main {

    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null;
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());

        int[] array = new int[n];
        for (int i = 0; i < n; i++) {
            array[i] = Integer.parseInt(st.nextToken());
        }
        int x = Integer.parseInt(br.readLine());

        Arrays.sort(array);

        int start = 0;
        int end = n-1;

        int count = 0;
        while (start < end) {
            int amount = array[start] + array[end];
            if (amount == x) {
                count++;
                end--;
            }
            else if (amount < x) {
                start++;
            }
            else
                end--;
        }
        System.out.println(count);
    }
}
